### DISCUSSION TITLE
Is it possible to use more than one repository in github page?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#58767] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I have a game mod, (The-modding-tree), and i want to use my github pages for my games. But games are different repositories. I wonder, if it's possible to acess different repositories from my github page. For example: i have my github page, i have repository with name "super-awesome-game" and that repository has index.html, and i have another repository with name "another-game" and with index.html too. And when i update them i should be able to share them at andryandryandry.github.io/<repositoryname>.

**Comment by anhtran917** [comment]
Yes, it is possible to access different repositories from your GitHub Pages website. 
I think you can follow these steps:

1. Create a new repository on GitHub with the name "andryandryandry.github.io". This will be the repository that hosts your GitHub Pages website.

2. Clone the "andryandryandry.github.io" repository to your local machine.
 
3. Create a new folder in the root directory of the "andryandryandry.github.io" repository for each game you want to host. 
    For example, you could create a folder called "super-awesome-game" and another folder called "another-game".

4. Clone each game repository into its corresponding folder in the "andryandryandry.github.io" repository.
 
5. Make sure each game repository has an "index.html" file in its root directory.
 
6. Commit and push the changes to the "andryandryandry.github.io" repository.

You should now be able to access each game at "andryandryandry.github.io/super-awesome-game" and "andryandryandry.github.io/another-game", respectively.

And you will need to update the "andryandryandry.github.io" repository each time you make changes to one of the game repositories.

**Reply by ghost** [reply]
Thank you! I hope this will help. Tho, i wonder, sometimes i see "enviroments: github pages" in repository, is this just a some sort of connection, OR this is something about custom domains?
![image](https://github.com/community/community/assets/127416891/0fe2b7ae-c8be-49c9-99da-22beefd9bae7)

**Comment by anhtran917** [comment]
Hello, Andry!
Well, "Environments: GitHub Pages" is a feature in GitHub that allows you to create a separate environment for your GitHub Pages website. This feature is not directly related to custom domains, but it can be used in conjunction with custom domains to create a more customized website.

When you create an environment for your GitHub Pages website, you can specify which branch of your repository should be used to build the website. This allows you to keep your source code and your website code separate, which can be useful if you have multiple contributors working on the same repository.

You can also use environments to test changes to your website before you push them to the live site. 
For example, you could create a "staging" environment that uses a separate branch of your repository, and then test your changes on the staging site before merging them into the main branch and pushing them to the live site.

So I think "Environments: GitHub Pages" is a powerful feature that can help you manage your GitHub Pages website more effectively. However, it is not necessary for hosting a basic website on GitHub Pages, and you can still use custom domains without using environments.

**Reply by ghost** [reply]
I wanted to ask is it possible to use enviroment not to test changes but to have site, but i will just search what "github enviroments" is lol. Thanks for really long and clear answers! Have a nice day! _Maybe i will still ask something here if i won't understand lol._

**Comment by anhtran917** [comment]
Sounds good, and it is possible to use an environment to host your GitHub Pages website. 
When you create an environment for your GitHub Pages website, you can specify which branch of your repository should be used to build the website. 
If you set the default branch of your repository to the branch you are using for your environment, then your website will be built automatically whenever you push changes to that branch.

Feel free to ask me more.

**Reply by ghost** [reply, marked as answer]
Lol, it was easier than i expect. I asked at XXX discord server and someone said:
1. Go to the repository where you want to have Github Pages
2. Go to the `Settings` tab
3. On the left, under `Code and automation`, go to `Pages`
4. Under `Build and deployment`, make sure `Source` is set to `Deploy from a branch`
5. In the same spot, set `Branch` to the branch where you have your mod
6. `Save` and you're done, it'll be up in just a minute
And it worked! Thank you for support, now i know how to use github pages for repository! Instead of using raw.githack! Yay!

**Reply by anhtran917** [reply]
I am glad to hear that you got a correct answer.
